docs/langref/src/concepts/purity.md

```diff
@@ -6,22 +6,25 @@ automatically discovered during property evaluation. The property stores the
 result of the evaluation. When a property changes, all dependent properties get
 notified, so that the next time their value is read, their binding is re-evaluated.
 
-For this reactive system to work well, bindings evaluation should be "pure":
-Evaluating a property shouldn't have side-effects. Side effects are
-problematic because it's not always clear when they will happen: Lazy evaluation
-may change their order or affect whether they happen at all. In addition, changes
-to properties during their binding evaluation due to a side-effect
-may result in unexpected behavior.
+For any reactive system to work well, bindings evaluation should be "pure":
+Evaluating a property shouldn't have side-effects on other properties or change
```

Comment:
I suppose the "properties" was added to clarification, but just today I was helping an ambassador with their code and in their C+ callback that's declared `pure` they were changing the rows of models. So I'd leave it "open" and not "limited" to properties here.

```suggestion
Evaluating a property shouldn't have side-effects and change
```